docs/static/breaking-changes.asciidoc

```diff
@@ -1,26 +1,202 @@
 [[breaking-changes]]
 == Breaking Changes
 
-We strive to maintain backward compatibility between minor versions (6.x to 6.y,
+We strive to maintain backward compatibility between minor versions (7.x to 7.y,
 for example) so that you can upgrade without changing any configuration files.
-Breaking changes are usually introduced only between major versions (such as 5.x
-to 6.y). On occasion, we are forced to break compatibility within a given major release
+Breaking changes are usually introduced only between major versions (such as 6.x
+to 7.y). On occasion, we are forced to break compatibility within a given major release
 to ensure correctness of operation.
 
 This section covers the changes that you need to be aware of when migrating to
-Logstash 6.0.0 and later.
+Logstash 7.0.0 and later.
 
-NOTE: Migrating directly between non-consecutive major versions (1.x to
-6.x) is not recommended.
+NOTE: Migrating directly between non-consecutive major versions (5.x to
+7.x) is not recommended.
 
 See these topics for a description of breaking changes:
 
+* <<breaking-7.0>>
 * <<breaking-pq>>
 * <<breaking-6.0>>
 
 See also <<releasenotes>>. 
 
+[[breaking-7.0]]
+=== Breaking changes in 7.0
+
+coming[7.0.0]
+
+Here are the breaking changes for {ls} 7.0. 
+
+[float]
+==== Changes in Logstash Core
+
+These changes can affect any instance of Logstash that uses impacted features.
+Changes to Logstash Core are plugin agnostic.
+
+[float]
+[[field-ref-strict]]
+===== Field Reference parser is more strict
+
+The Field Reference parser, which is used to interpret references to fields in
+your pipelines and plugins, was made to be more strict and will now reject
+inputs that are either ambiguous or illegal. Since 6.4, Logstash has emitted
+warnings when encountering input that is ambiguous, and allowed an early opt-in
+of strict-mode parsing either by providing the command-line flag
+`--field-reference-parser STRICT` or by adding `config.field_reference.parser:
+STRICT` to `logstash.yml`.
```

Comment:
can we maybe have an example here? not sure if it will be too verbose:

Before:

```
logstash-6.7.0 % echo "hello"| bin/logstash -e 'filter { mutate { replace => { "message" => "%{[[]]message]} you" } } }'
[2019-04-05T16:52:18,691][WARN ][org.logstash.FieldReference] Detected ambiguous Field Reference `[[]]message]`, which we expanded to the path `[message]`; in a future release of Logstash, ambiguous Field References will not be expanded.
{
       "message" => "hello you",
      "@version" => "1",
    "@timestamp" => 2019-04-05T15:52:18.546Z,
          "type" => "stdin",
          "host" => "overcraft.lan"
}
```

After:

```
logstash-7.0.0 % echo "hello"| bin/logstash -e 'filter { mutate { replace => { "message" => "%{[[]]message]} you" } } }'
[2019-04-05T16:48:09,135][FATAL][logstash.runner          ] An unexpected error occurred! {:error=>java.lang.IllegalStateException: org.logstash.FieldReference$IllegalSyntaxException: Invalid FieldReference: `[[]]message]`
[2019-04-05T16:48:09,167][ERROR][org.logstash.Logstash    ] java.lang.IllegalStateException: Logstash stopped processing because of an error: (SystemExit) exit
```